src/extension.ts

```diff
@@ -1,21 +1,77 @@
-import * as vscode from 'vscode';
+import { exec } from 'child_process';
+import * as fs from 'fs/promises';
 import * as os from 'os';
 import * as path from 'path';
-import { exec } from 'child_process';
 import { promisify } from 'util';
+import * as vscode from 'vscode';
 import {
+    Executable,
     LanguageClient,
     LanguageClientOptions,
     ServerOptions,
-    TransportKind,
-    Executable
+    TransportKind
 } from 'vscode-languageclient/node';
+import which from 'which';
 
 let client: LanguageClient;
 
 const execAsync = promisify(exec);
 const MIN_ZIZMOR_VERSION = '1.11.0';
 
+async function getZizmor(config: vscode.WorkspaceConfiguration): Promise<string | undefined> {
+    const rawConfiguredPath = config.get<string>('executablePath');
+
+    if (rawConfiguredPath) {
+        const configuredPath = await which(expandTilde(rawConfiguredPath), { nothrow: true });
+
+        if (configuredPath) {
+            console.log(`Using configured zizmor: ${configuredPath}`);
+            return configuredPath;
+        } else {
+            console.warn(`Configured Zizmor not found: ${rawConfiguredPath}`)
+        }
+    }
+
+    const pathPath = await which('zizmor', { nothrow: true })
+    if (pathPath) {
+        console.log(`Using Zizmor from PATH: ${pathPath}`)
+        return pathPath;
+    }
+
+
+    // It's particularly important to check common locations on macOS because of https://github.com/microsoft/vscode/issues/30847#issuecomment-420399383.
+    const commonPathsLocal = [
+        path.join(os.homedir(), ".cargo", "bin", "zizmor"),
+        path.join(os.homedir(), ".cargo", "bin", "zizmor.exe"),
+        path.join(os.homedir(), ".nix-profile", "bin", "zizmor"),
+        path.join(os.homedir(), ".local", "bin", "zizmor"),
+        path.join(os.homedir(), "bin", "zizmor"),
+    ]
+    const commonPathsUnixGlobal = [
+        "/usr/bin/zizmor",
+        "/home/linuxbrew/.linuxbrew/bin/zizmor",
+        "/usr/local/bin/zizmor",
+        "/opt/homebrew/bin/zizmor",
+        "/opt/local/bin/zizmor",
+    ];
+
+    // Windows supports forward slashes, but these paths won't work without drive letters.
+    // I figure old DOSes we don't support might not support forward slashes, so this is a good enough check.
+    const commonPaths = path.sep == "/" ? [...commonPathsLocal, ...commonPathsUnixGlobal] : commonPathsLocal
+
+    for (const commonPath of commonPaths) {
+        try {
+            await fs.access(commonPath, fs.constants.X_OK)
```

Comment:
Not sure if `X_OK` will work on Windows... might need to handle PATHEXT and at that point we should just whip out `which` again.

Reply:
Hmm, do shells actually do `X_OK`? I was under the impression they only looked for a matching name in the path set, and allowed `EACCESS` to percolate.

Here's what I see on macOS for example:

```
tmp % bingo
zsh: command not found: bingo
tmp % echo $PWD
/Users/william/tmp
tmp % touch bingo
tmp % export PATH=$PWD:$PATH
tmp % bingo
zsh: permission denied: bingo
```

(That's with zsh, but I expect bash would do the same.)

Reply:
My understanding is that they look for the first executable, if there isn't an executable, they "perm denied", and if there isn't any match at all, they "not found".

Could be wrong, I'll look into it.

Reply:
For reference, _this_ code doesn't exist anymore it's in `which.ts` now... and thinking about consistency which which explained it. `which` (at least the one from macOS builtin zsh) _does_ factor in executable bits, but the lookup (at least on new, non-macOS) bash does not (as linked above). Apparently that's one of the reasons `command -v` is recommended these days.